### ISSUE TITLE
FOR expression doesn't work correctly with non-trivial expression

### ISSUE THREAD

**Issue by vpetrovykh** [edgedb/edgedb#1283]
- EdgeDB Version: 1.0-alpha.3+dev.82.ga77f6950
- OS Version: Linux

Steps to Reproduce:
```
db> CREATE TYPE Foo {CREATE PROPERTY q -> str {CREATE CONSTRAINT exclusive}};
CREATE
db> FOR x IN {2,3} UNION (INSERT Foo {q := 'q' ++ <str>x});
ConstraintViolationError: q violates exclusivity constraint
db> FOR x IN {2,3} UNION (SELECT Object{q := 'q' ++ <str>x} LIMIT 1);
{Object { q: 'q(2, 3)' }, Object { q: 'q(2, 3)' }}
db> FOR x IN {2, 3} UNION (SELECT 'q' ++ <str>x);
{'q(2, 3)', 'q(2, 3)'}
db> FOR x IN {2, 3} UNION (SELECT x);
{2, 3}
```

**Comment by 1st1** [comment]
@elprans Can we fix this for alpha 3? If not please move to alpha 4.

**Comment by elprans** [comment]
I cannot reproduce this anymore, so this likely got fixed.  @vpetrovykh is there a test for this.

**Comment by vpetrovykh** [comment]
There's a similar in structure `FOR` test (`test_edgeql_for_in_computable_02`) that was fixed in d9429f5fe56e40a85c2665d6d1107ab978309d18. It's a more complicated `FOR`, but it would fail if the examples from this ticket didn't work.
